packit/upstream.py

```diff
@@ -179,22 +180,55 @@ def commit(self, title: str, msg: str, prefix: str = "[packit] ") -> None:
             commit_args += ["-m", msg]
         self.local_project.git_repo.git.commit(*commit_args)
 
-    def push_to_branch(
-        self, branch_name: str, remote_name: str = "origin", force: bool = False
-    ):
+    def push(self, branch_name: str, force: bool = False, fork: bool = True) -> str:
         """
-        push changes to a fork of the dist-git repo; they need to be committed!
+        push current branch to fork if fork=True, else to origin
 
         :param branch_name: the branch where we push
-        :param fork_remote_name: local name of the remote where we push to
         :param force: push forcefully?
+        :param fork: push to fork?
+        :return: name of the branch where we pushed
         """
-        # I suggest to comment this one while testing when the push is not needed
-        # TODO: create dry-run ^
-        self.local_project.git_repo.remote(remote_name).push(
-            refspec=branch_name, force=force
+        logger.debug(
+            f"About to {'force ' if force else ''}push changes to branch {branch_name}."
         )
 
+        if fork:
+            if self.local_project.git_project.is_fork:
+                fork_project = self.local_project.git_project
+            else:
+                fork_project = self.local_project.git_project.get_fork(create=True)
```

Comment:
Are we going to eventually utilize `get_fork()` from packit-service/ogr/pull/44 ?

Reply:
I would be nice to wait for the merge of that PR. (Ideally, release OGR after that.)

Reply:
@jpopelka that IS the get_fork method